source/guide/3.2/plugin-aws.md

```diff
@@ -0,0 +1,219 @@
+
+
+## Runtime Properties
+
+Node instances of any of the types defined in this plugin get set with the following runtime properties during the `cloudify.interfaces.lifecycle.create` operation:
+
+  * `aws_resource_id` the AWS ID of the resource
+
+
+## Default Resource Naming Convention
+
+If 'use_external_resource' is set to true in the blueprint, the 'resource_id' must be that resource's ID in AWS, unless the resource type is a keypair, in which case the value is the key's name.
+
+
+
+# Using Existing Resources
+
+It is possible to use existing resources on AWS - whether these have been created by a different Cloudify deployment or not via Cloudify at all.
+
+All Cloudify AWS types have a property named `use_external_resource`, whose default value is `false`. When set to `true`, the plugin will apply different semantics for each of the operations executed on the relevant node's instances:
+
+This behavior is common to all resource types:
+
+ * `create` If use_external_resource is true, the AWS plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will assign the resource id to the instance runtime_properties.
+
+ * `delete` If use_external_resource is true, the AWS plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will unassign the instance runtime_properties.
+
+The following behaviors are unique:
+
+ * `aws.ec2.instance.start` If use_external_resource is true, the runtime properties for public_ip_address, etc, are set, and the function exits.
+ * `aws.ec2.instance.stop` If use_external_resource is true, the runtime properties for public_ip_address, etc, are unset, and the function exits.
+ * `cloudify.aws.relationships.instance_connected_to_elastic_ip` Here, both the instance's and the elasticip's use_external_resource value are relevant. If both are external the function sets the relationship properties. If either are not external the function fails.
+ * `cloudify.aws.relationships.instance_connected_to_security_group` Here, both the instance's and the security groups's use_external_resource value are relevant. If both are external the function sets the relationship properties. If either are not external the function fails.
+
+
+# Account Information
+
+* The plugin needs access do your aws_access_key_id and aws_secret_access_key in order to operate.
+
+Please read about your AWS Boto configuration here:  http://boto.readthedocs.org/en/latest/boto_config_tut.html
+
+
+
+# Tips
+
+* It is highly recommended to **ensure that AWS names are unique.
+
+
+
+# Misc
+
+* Some configuration-saving information is available in the [Provider Context](reference-terminology.html#provider-context).
+
+  The exact details of the structure of the AWS Provider Context are not documented, because parts may change.
```

Comment:
Is this informative enough? This should certainly contain a reference to the terminology and explained so that advanced users understand how they can use the provider context.